python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/updated_scenarios/partition_scenarios.py

```diff
@@ -742,4 +745,25 @@
         .evaluate_tick()
         .assert_requested_runs(),
     ),
+    # Ensure that with a rule that skips on run in progress, we don't launch a new run
+    AssetDaemonScenario(
+        id="partitioned_asset_in_progress-rule",
+        initial_spec=two_assets_in_sequence.with_asset_properties(
+            keys=["B"],
+            partitions_def=one_partitions_def,
+            auto_materialize_policy=AutoMaterializePolicy.eager().with_rules(
+                SkipOnRunInProgressRule(all_partitions=True)
+            ),
+        ),
+        execution_fn=lambda state: state.with_runs(run_request(["A"]))
+        .evaluate_tick()
+        .assert_requested_runs(run_request(["B"]))
+        .with_not_started_runs()
+        .evaluate_tick()
+        .assert_requested_runs()
+        .with_runs(run_request(["A"]))
+        .with_in_progress_run_for_asset("B")
```

Comment:
somewhat surprised that this call doesn't fail given the fact that B is partitioned. I think you'd ideally want to update that method to be:

```python
    def with_in_progress_run_for_asset(
        self, asset_key: CoercibleToAssetKey, partition_key: Optional[str] = None
    ) -> Self:
        in_progress_run_id = make_new_run_id()
        with pendulum_freeze_time(self.current_time):
            asset_key = AssetKey.from_coercible(asset_key)
            job_def = self.scenario_spec.defs.get_implicit_job_def_for_assets(
                asset_keys=[asset_key]
            )
            assert job_def
            execution_plan = create_execution_plan(job_def, run_config={})
            self.instance.create_run_for_job(
                job_def=job_def,
                run_id=in_progress_run_id,
                status=DagsterRunStatus.STARTED,
                asset_selection=frozenset({AssetKey.from_coercible(asset_key)}),
                execution_plan=execution_plan,
                tags={PARTITION_NAME_TAG: partition_key} if partition_key else None,
            )
            assert self.instance.get_run_by_id(in_progress_run_id)
        return self

```

Reply:
I don't think I was able to run the tests. thank you - I'll fix it.